### PR TITLE
Fixed `MongoDBCollection#watch` on React Native

### LINKED ISSUE #3494: collection.watch() fails on React Native SDK
## Goals
I'm trying to monitor collection changes on a remote MongoDB Atlas database.

## Expected Results
I followed the example on the [official](https://docs.mongodb.com/realm/react-native/mongodb#std-label-react-native-mongodb-watch) Realm docs to use the collection.watch() functionality for the RN SDK. The example works with Node, and it returns any newly inserted objects in the collection, which is what I'm going for. The exact same code, however, does not work in my RN project.

## Actual Results
I get the following error message

```
EROR [TypeError: Invalid attempt to iterate non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.]
```

## Steps to Reproduce
The error can be reproduced running the following code with the specified SDK and RN versions at the bottom. I call this function inside the `componentDidMount()` function, which looks like this:

 ```js
 componentDidMount() {
    this.watchMessageCollection().catch((err) => console.log('ERROR', err));
  }
 ```
And watchMessageCollection():

```js
  async watchMessageCollection() {
    const {user} = this.props.session;
    const db = user.mongoClient('mongodb-atlas').db('main');
    const messages_collection = db.collection('chat_messages');
    const generator = messages_collection.watch();

    for await (const change of generator) {
      const {documentKey, fullDocument} = change;
      console.log(`new document: ${documentKey}`, fullDocument);
    }
  }

```
I tried running this function on different project files with the same result, so there's something about the generator itself that's off. The user auth works on my RN project, as do the database and collection connections.

## Version of Realm and Tooling

- Realm JS SDK Version: 10.1.2
- Node: v15.1.0
- React Native: 0.63.4
- OS: Ubuntu 20.04.1 LTS

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This closes #3494 by adding documentation on installing polyfills and a babel plugin, plus calling the React Native fetch polyfill with the right option.

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests (manually locally)
* [x] 📱 Check the React Native/other sample apps work if necessary
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "./header.eslintrc"
   ],
   "env": {
-    "es2017": true
+    "es2020": true
```
Comment (kraenhansen) [inline .eslintrc.json:9]: This is needed to be able to use the `globalThis` keyword.

File: packages/realm-network-transport/rollup.config.js [hunk 2]
```diff
@@ -63,6 +63,21 @@ export default [
       nodeResolve(),
     ],
   },
+  {
+    input: "src/react-native/index.ts",
```
Comment (kraenhansen) [inline packages/realm-network-transport/rollup.config.js:67]: We're adding a new "react-native" bundle to separate the code that runs on "react-native" from that which run on web.

File: packages/realm-network-transport/src/react-native/index.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import { DefaultNetworkTransport } from "../DefaultNetworkTransport";
+import { AbortController, Fetch } from "../types";
+
+DefaultNetworkTransport.fetch = globalThis.fetch.bind(globalThis) as Fetch;
+DefaultNetworkTransport.AbortController = globalThis.AbortController as AbortController;
+// Setting this non-standard option to enable text streaming
+// See https://github.com/react-native-community/fetch#enable-text-streaming
+DefaultNetworkTransport.extraFetchOptions = {
+  reactNative: { textStreaming: true },
```
Comment (kraenhansen) [inline packages/realm-network-transport/src/react-native/index.ts:29]: See https://github.com/react-native-community/fetch#enable-text-streaming for details on why we're adding this.

File: packages/realm-network-transport/src/DefaultNetworkTransport.ts [hunk 4]
```diff
@@ -69,6 +70,7 @@ export class DefaultNetworkTransport implements NetworkTransport {
     try {
       // We'll await the response to catch throw our own error
       return await DefaultNetworkTransport.fetch(url, {
+        ...DefaultNetworkTransport.extraFetchOptions,
```
Comment (kraenhansen) [inline packages/realm-network-transport/src/DefaultNetworkTransport.ts:73]: I tried setting this only on the streaming requests, but that yield the following in my test app:

```
 LOG  Running "ReactNativeTestApp" with {"rootTag":61,"initialProps":{}}
 WARN  Invalid responseType: blob
 ERROR  {"code": -1, "message": "undefined is not an object (evaluating 'options.blobId')"}
 WARN  Invalid responseType: blob
 LOG  Unable to symbolicate stack trace: undefined is not an object (evaluating 'options.blobId')
 WARN  Invalid responseType: blob
 WARN  Invalid responseType: blob
 WARN  Invalid responseType: blob
```

File: packages/realm-network-transport/src/react-native/index.ts (hunk 3 above)
Comment (kneth) [reply packages/realm-network-transport/src/react-native/index.ts:29]: Please add it as a comment

**Review by kneth — APPROVED** [review]

**Review by takameyer — APPROVED** [review]
Awesome.  Do we need to prominently document this somewhere, besides in the comments?

File: packages/realm-network-transport/src/react-native/index.ts (hunk 3 above)
Comment (kraenhansen) [reply packages/realm-network-transport/src/react-native/index.ts:29]: It's already there - two lines above the line I commented on here :)

**Comment by kraenhansen** [comment]
> Awesome. Do we need to prominently document this somewhere, besides in the comments?

Besides the changes I'm proposing to the TypeScript docs and JSDoc, I would say the instructions on adding the babel plugin and globals polyfill needs to be inserted on https://docs.mongodb.com/realm/sdk/react-native/examples/query-mongodb/#real-time-change-notifications as well.

**Pull request merged**